src/transform-sdk/rust/core-sys/src/lib.rs

```diff
@@ -0,0 +1,137 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{
+    fmt::Debug,
+    time::{Duration, SystemTime},
+};
+
+#[cfg(target_os = "wasi")]
+mod abi;
+#[cfg(not(target_os = "wasi"))]
+mod stub_abi;
+#[cfg(not(target_os = "wasi"))]
+use stub_abi as abi;
+mod serde;
+mod varint;
+
+extern crate redpanda_transform_sdk_types;
+
+use redpanda_transform_sdk_types::*;
+
+#[cfg(test)]
+#[macro_use]
+extern crate quickcheck;
+
+#[cfg(test)]
+extern crate rand;
+
+pub fn process<E, F>(cb: F)
+where
+    E: Debug,
```

Comment:
suggestion: bind `E` to `std::error::Error`

Reply:
Let's centralize this to https://github.com/redpanda-data/redpanda/pull/14857#discussion_r1411571626